test/test_testing.py

```diff
@@ -2781,56 +2781,127 @@ def test_add_layer(self):
-    description: |
-      A description about Serv the amazing service.
-    startup: enabled
-    override: replace
-    command: '/bin/echo hello'
-    environment:
-      KEY: VALUE
-'''))
+            summary: Foo
+            description: |
+              A longer description about Foo
+            services:
+              serv:
+                summary: Serv
+                description: |
+                  A description about Serv the amazing service.
+                startup: enabled
+                override: replace
+                command: '/bin/echo hello'
+                environment:
+                  KEY: VALUE
+            '''))
+        plan = client.get_plan()
+        # The YAML should be normalized
+        self.assertEqual(textwrap.dedent('''\
+            services:
+              serv:
+                command: /bin/echo hello
+                description: 'A description about Serv the amazing service.
+
+                  '
+                environment:
+                  KEY: VALUE
+                override: replace
+                startup: enabled
+                summary: Serv
+            '''), plan.to_yaml())
+
+    def test_add_layer_merge(self):
+        client = self.get_testing_client()
+        plan = client.get_plan()
+        self.assertIsInstance(plan, pebble.Plan)
+        self.assertEqual('{}\n', plan.to_yaml())
+        client.add_layer('foo', pebble.Layer('''\
+            summary: Foo
+            description: |
+              A longer description about Foo
+            services:
+              serv:
+                summary: Serv
+                description: |
+                  A description about Serv the amazing service.
+                startup: enabled
+                override: replace
+                command: '/bin/echo hello'
+                environment:
+                  KEY: VALUE
+            '''))
+        plan = client.get_plan()
+        # The YAML should be normalized
+        self.assertEqual(textwrap.dedent('''\
+            services:
+              serv:
+                command: /bin/echo hello
+                description: 'A description about Serv the amazing service.
+
+                  '
+                environment:
+                  KEY: VALUE
+                override: replace
+                startup: enabled
+                summary: Serv
+            '''), plan.to_yaml())
+
+        client.add_layer('foo', pebble.Layer('''\
+            summary: Foo
+            description: |
+              A longer description about Foo
+            services:
+              serv:
+                summary: Serv
+                description: |
+                  A new description of the the amazing Serv service.
+                startup: enabled
+                override: merge
+                command: '/bin/echo hello'
+                environment:
+                  KEY: VALUE1
+                  KEY1: VALUE2
+                  KEY3: VALUE3
+            '''), combine=True)
```

Comment:
I think we should probably have one test here that has all fields (per above, we happened to miss the buggy fields here :-).

Reply:
added fields